pennylane/_qubit_device.py

```diff
@@ -119,44 +126,96 @@ def execute(self, queue, observables, parameters=None):
 
             return self._asarray(results)
 
-    def rotate_basis(self, observable):
+    def rotate_basis(self, obs_queue):
         """Rotates the specified wires such that they
         are in the eigenbasis of the provided observable.
+        """
+        self._prob = self.probability(values_only=True)
```

Comment:
We had a discussion about this attribute, could you remind me what the devices could use this attribute for?

Reply:
This is a bit hacky, and it's only used to avoid creating that ordered dictionary (which would increase memory usage for very large systems)

Reply:
Ideally, we would implement a better return type for `probability`, such as a custom class.

Reply:
Provided that the ``basis_states`` are always created based on the order coming from
```
        basis_states = itertools.product(range(2), repeat=len(wires))
```
(which could be documented), we could actually always just return ``probability``, correct?

The user can also just construct ``basis_states`` or we could even have a ``property`` for that such as
```python
@property
def probabilitiy_with_basis_states(self):
    basis_states = itertools.product(range(2), repeat=len(wires))
    return dict(zip(basis_states, self.probability))
```
This can be done like this, as [``itertools.product``](https://docs.python.org/3.8/library/itertools.html#itertools.product) keeps ordering:
> Roughly equivalent to nested for-loops in a generator expression. For example, product(A, B) returns the same as ((x,y) for x in A for y in B).